src/dir/glob.cr

```diff
@@ -223,7 +223,7 @@ class Dir
 
     private def self.constant_entry?(file)
       file.each_char do |char|
-        return false if char.in?('*', '?')
+        return false if char.in?('*', '?', '{', '[', '\\')
```

Comment:
There shouldn't be a need for `{`, because `File.expand_brace_pattern` is always called before this method, which means `file` can never contain unescaped `{`s (the escaped ones will have been caught by the `\\`).